compiler/rustc_feature/src/builtin_attrs.rs

```diff
@@ -338,9 +342,11 @@ pub const BUILTIN_ATTRIBUTES: &[BuiltinAttribute] = &[
         template!(List: r#"name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated""#),
         DuplicatesOk,
     ),
-    ungated!(link_name, Normal, template!(NameValueStr: "name"), FutureWarnPreceding),
+    ungated!(
+        link_name, Normal, template!(NameValueStr: "name"), FutureWarnPreceding, @only_local: true,
```

Comment:
From test failure, miri appears to need this attribute.

Reply:
Absolutely, we're currently investigating it.

Reply:
Yes [it uses it](https://github.com/rust-lang/miri/blob/3e911255374def9fbe142160da36e688b05fea13/src/helpers.rs#L880) opportunistically and can handle its absence, so we could remove it without making it local_only. But we'll need to see how the `name` is used in the interpreter, e.g. in the known functions shims, before doing that.

Reply:
It is used to resolve `extern "C" { fn ... }` to `#[no_mangle] extern "C" fn ...` as well as to resolve it to function shims I believe.

Reply:
Yeah Miri needs to know the link_name to resolve `extern` function calls. The fallback prevents an ICE but is incorrect if the function is actually supposed to have another link_name.

If removing link_name is somehow a big win, Miri would need a way to opt-out from that via some `-Z` flag.

Reply:
I have put the new numbers [here](https://github.com/rust-lang/rust/pull/98450#issuecomment-1275187390). Up to you to see if it's worth the trouble.